dags/top-public-synapse-projects-from-snowflake.py

```
@@ -152,72 +217,408 @@ def get_top_downloads_from_snowflake(**context) -> List[DownloadMetric]:
+            for _, row in public_df.iterrows():
+                metrics.append(
+                    DownloadMetric(
+                        name=row["NAME"],
+                        project=row["PROJECT"],
+                        downloads_per_project=row["DOWNLOADS_PER_PROJECT"],
+                        number_of_unique_users_downloaded=row[
+                            "NUMBER_OF_UNIQUE_USERS_DOWNLOADED"
+                        ],
+                        data_download_size=row["DATA_DOWNLOAD_SIZE"] or 0,
+                    )
                 )
+        finally:
+            cs.close()
+
+        return metrics
+
+    @task
+    def get_fileview_downloads_from_snowflake(**context) -> List[DownloadMetric]:
+        """Execute a query on Snowflake and return download metrics for fileview groups.
+
+        This function executes a single query that:
+        1. Parses the fileview_groups parameter to get file view IDs and group names
+        2. Uses the file view scopes to identify projects belonging to each group
+        3. Gets download stats for those projects, grouped by their fileview groups
+        4. Allows projects to appear in multiple groups if they are in multiple fileviews
+
+        Arguments:
+            context: Airflow context dictionary containing DAG parameters
+                - snowflake_developer_service_conn: Connection ID for Snowflake
+                - fileview_groups: JSON string of fileview ID and group name mappings
+                - backfill_date: Date string for backfill in YYYY-MM-DD format
+                - hours_time_delta: Hours to subtract from current/backfill date
+                - backfill: Boolean indicating whether this is a backfill run
+
+        Returns:
+            List[DownloadMetric]: A list of download metrics for fileview groups
+        """
+        try:
+            groups = json.loads(context["params"]["fileview_groups"])
+        except json.JSONDecodeError:
+            return []
```

Would we not want to raise an error saying that the string was malformed?